src/Auth/OAuth.js

```diff
@@ -0,0 +1,18 @@
+class OAuth {
```

Comment:
This object contains data from 2 different entities:

1. Data about the access token (this._accessToken)
2. Data referring to the authentication (this._type)

I would split this in 2 objects:

1. Authentication (where you have type and configuration) - The TokenResponse with little tweaks can be turned into this
2. AccessToken: this would hold the relevant information about the access token see [HERE](https://github.com/thephpleague/oauth2-client/blob/master/src/Token/AccessToken.php)